### DISCUSSION TITLE
如何自定义数据库的 schema？(How to customize Schema of database?)

### DISCUSSION THREAD

**Discussion by hooozen** [hexojs/hexo#5124] [Question / Answer] [answered]
I want to add a model like CategoryTag which would perform like PostCategory or PostTag. In the source code, The PostCategory and PostTag model is set when processing post files. But I needn't process files again. so are there any hooks or methods to let users customize schema?

Hexo（或 warehouse） 是否提供了合适的方法或钩子，来修改 Schema，或者自定义 Schema。能够使我能在所有的文件资源被 Process 完毕后，利用 Post 和 Categories 等 model 来自定义我自己的 Model 呢？并且能够在 Generator 被执行前完成。

我现在找不到一个合适的时机：若在 ProcessAfter 后执行，由于数据库的插入等操作是异步执行的，那么 Generator 就读取不到正确的数据了。如果在 ProcessBefor 前执行，那么我就得自己处理资源文件，而无法利用 Hexo 已经为我们构建好的 Post 和 Categories 数据。

在现的 Hexo 有没有很好的解决方法呢？

**Comment by stevenjoezhang** [comment, marked as answer]
按照目前的设计并不允许第三方插件自定义 Model。
